### PR TITLE
DAOS-2167 test: restore/simplify pool destroy flow in tests

### PR DESCRIPTION
**Pull request by kccain**
With this change, temporary measures added in PR #1643 adding a pool
disconnect call before destroy(force=true) are restored to the original
form. This is possible due to recent fixes to DAOS pool destroy in
PR #1625. Also with this change, a centralized function to disconnect
from and remove pools used by a test specifies force=false when calling
the DAOS pool destroy API.

The pool bad_evict and bad_exclude test yaml configuration files
are updated so the agent and the daos_server configuration both use
the non-default DAOS system name (only daos_server used the name
causing the test to not run properly).

The container_async test is updated to pass an additional argument
that was added to a container create python function. The test would
not run properly prior to this change.

Test-tag: pr,-hw badevict badexclude container rebuildcontcreate object
Test-tag-hw: pr,hw

Signed-off-by: Ken Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

**Comment by kccain** [comment]
@rpadma2  and @phender  this is now ready for your review. Passed functional testing. The 4 test failures are identical to the test failure signatures seen in those same tests a recent weekly full_regression test run (build #54 of the weekly-testing branch).

File: src/tests/ftest/container/attribute.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ hosts:
   test_servers:
     - server-A
 timeout: 50
+server_manager:
+  recreate: true
```
Comment (rpadma2) [inline src/tests/ftest/container/attribute.yaml]: Does this add --recreate-superblocks to command line? Aren't we supposed to remove them and start performing --reformat for both tmpfs and dcpm?

**Comment by rpadma2** [comment]
Any reason we are encountering functional test failures?

File: src/tests/ftest/container/attribute.yaml (hunk 1 above)
Comment (kccain) [reply src/tests/ftest/container/attribute.yaml]: Yes I believe it does. It is a consequence of this very recent PR that now requires all tests that use recreate-superblocks on server start to specifically opt-in
https://github.com/daos-stack/daos/commit/81a26c4987732b6a19f19f26516a82b8fee74f31

It is not a goal of this patch to fix the infrastructure for launching tests. There is another PR in progress as I understand to address server launch issues.

**Comment by kccain** [comment]
> Any reason why we are encountering functional test failures?

Yes I noted this above when I requested the review
"The 4 test failures are identical to the test failure signatures seen in those same tests a recent weekly full_regression test run (build #54 of the weekly-testing branch)."

Those problems need to be tracked down separately from this PR.

**Comment by kccain** [comment]
A couple of notes on the latest commit functional testing failures:

1. The 4 test failures are identical to the test failure signatures seen in those same tests a recent weekly full_regression test run (build #54 of the weekly-testing branch). The 4 test failures being:

1-./container/container_async.py:ContainerAsync.test_createasync;mode_RO-createset-createsize-hosts-server_config-ad94 – ContainerAsync
24s
2-./container/create.py:CreateContainerTest.test_container_create;hosts-badone-poolparams-server_config-perfect-f5b4 – CreateContainerTest
22s
2-./object/obj_fetch_bad_param.py:ObjFetchBadParam.test_null_ptrs;hosts-createmode-createset-createsize-server_config-e5d8 – ObjFetchBadParam
29s
2-./object/obj_update_bad_param.py:ObjUpdateBadParam.test_null_values;createmode-createset-createsize-hosts-server_config-4129 – ObjUpdateBadParam

2. Merging with master to get the test infrastructure changes from https://github.com/daos-stack/daos/pull/1615 (https://jira.hpdd.intel.com/browse/DAOS-3914) seems to have resulted in some additional failures - dozens of test cases reporting errors similar to this:
"'BadExcludeTest' object has no attribute 'agent_log'"

I have added a note to DAOS-3914 with the observation.

**Comment by kccain** [comment]
Abandoning this PR - too many refactorings have occurred with the test infrastructure such that the purpose of this PR is no longer applicable.

A better approach will be to add a new test to destroy_tests.py tests to exercise pool destroy (with force=1) while the pool is connected. There is a negative test that destroys a connected pool (with force=0), expecting an exception/failure.

**Pull request closed without merging**